pkg/api/gitops/v1alpha1/application.go

```diff
@@ -17,12 +17,215 @@ limitations under the License.
 package v1alpha1
 
 import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	helmv2 "kubesphere.io/devops/pkg/external/fluxcd/helm/v2beta1"
+	kusv1 "kubesphere.io/devops/pkg/external/fluxcd/kustomize/v1beta2"
+	"kubesphere.io/devops/pkg/external/fluxcd/meta"
 )
 
+type FluxApplication struct {
+	Spec FluxApplicationSpec `json:"spec,omitempty"`
+}
+
+type FluxApplicationSpec struct {
+	Source *FluxApplicationSource `json:"source,omitempty"`
+	Config *FluxApplicationConfig `json:"config"`
+}
+
+type FluxApplicationSource struct {
+	SourceRef helmv2.CrossNamespaceObjectReference `json:"sourceRef"`
+}
+
+type FluxApplicationDestination struct {
+	KubeConfig *helmv2.KubeConfig `json:"kubeConfig,omitempty"`
+	// TargetNamespace to target when performing operations for the HelmRelease.
+	// Defaults to the namespace of the HelmRelease.
+	TargetNamespace string `json:"targetNamespace,omitempty"`
+}
+
+type FluxApplicationConfig struct {
```

Comment:
The comment on all the exported types is necessary. I don't know why the lint does not find it.